packages/cli/src/utils/contracts.js

```diff
@@ -26,3 +21,17 @@ export function bytecodeDigest(rawBytecode) {
 export function flattenSourceCode(contract) {
   return flatten(contract)
 }
+
+function splitCode(instance) {
+  const bytecode = instance.constructor.bytecode.replace(/^0x/, '')
+  const body = instance.constructor.deployedBytecode.replace(/^0x/, '')
+  const constructor = bytecode.substr(0, bytecode.indexOf(body))
+  return { constructor, body }
+}
+
+// Removes the last 43 bytes of the bytecode, i.e., the swarm hash that the solidity compiler appends and that
+// respects the following structure: 0xa1 0x65 'b' 'z' 'z' 'r' '0' 0x58 0x20 <32 bytes swarm hash> 0x00 0x29
+// (see https://solidity.readthedocs.io/en/v0.4.24/metadata.html#encoding-of-the-metadata-hash-in-the-bytecode)
+function removeSwarmHash(bytecode) {
+  return bytecode.replace(/a165627a7a72305820.*0029$/, '')
```

Comment:
Let's assert that the swarm hash is indeed 32 bytes. We can replace `.*` in the regex for `[a-fA-F0-9]{64}`.

Reply:
Also, it'd be interesting to fail if the replace operation does not match anything to replace, in case we stumble upon a contract with a different swarm hash format.

Reply:
@spalladino what do you mean by "fail"?

Reply:
already discussed it offline.